sky/data/sky_csync.py

```diff
@@ -0,0 +1,545 @@
+    # as initial backoff
+    backoff = common_utils.Backoff(int(interval_seconds / 2))
+    for i in range(_MAX_SYNC_RETRIES):
+        with subprocess.Popen(sync_cmd,
+                              start_new_session=True,
+                              shell=True,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE,
+                              text=True) as sync_process:
+            _set_running_csync_sync_pid(csync_pid, sync_process.pid)
+            stdout, stderr = sync_process.communicate()
+            _set_running_csync_sync_pid(csync_pid, -1)
+            sys.stdout.write(stdout)
+            sys.stdout.flush()
+            rc = sync_process.returncode
+            if rc != 0:
+                wait_time = backoff.current_backoff()
+                logger.warning('Encountered an error while syncing '
+                               f'{source_to_bucket}. Retrying sync '
+                               f'in {wait_time}s. {_MAX_SYNC_RETRIES-i-1} more '
+                               'reattempts remaining. Detailed error: '
+                               f'{stderr}')
+                time.sleep(wait_time)
+                continue
+            # successfully completed sync process.
+            return
+    raise RuntimeError(f'Failed to sync {source_to_bucket} after '
+                       f'{_MAX_SYNC_RETRIES} number of retries. Check '
+                       'the log file in ~/.sky/ for more'
+                       'details') from None
+
+
+def get_storage_mount_script(storetype: str, destination: str,
+                             mount_path: str) -> str:
+    if storetype == 's3':
+        install_cmd = mounting_utils.get_s3_mount_install_cmd()
+        mount_cmd = mounting_utils.get_s3_mount_cmd(destination, mount_path)
+    else:  # storetype == 'gcs':
+        install_cmd = mounting_utils.get_gcs_mount_install_cmd()
+        mount_cmd = mounting_utils.get_gcs_mount_cmd(destination, mount_path)
+
+    storage_mount_script = mounting_utils.get_mounting_script(
+        StorageMode.MOUNT, mount_path, mount_cmd, install_cmd)
+    return storage_mount_script
+
+
+def _handle_fuse_process(fuse_cmd: str) -> Tuple[int, int, str]:
+    with subprocess.Popen(fuse_cmd,
+                          shell=True,
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE,
+                          text=True) as fuse_process:
+        fuse_pid = fuse_process.pid
+        stdout, stderr = fuse_process.communicate()
+        rc = fuse_process.returncode
+        sys.stdout.write(stdout)
+        sys.stdout.flush()
+        stderr = stdout + stderr
+    if rc != 0:
+        stderr = stdout + stderr
+        sys.stderr.write(f'FUSE error: {stderr}')
+        sys.stderr.flush()
+    return fuse_pid, rc, stderr
+
+
+@main.command()
+@click.argument('source', required=True, type=str)
+@click.argument('storetype', required=True, type=str)
```

Comment:
this should be a click.Choice? ref: https://github.com/skypilot-org/skypilot/blob/553082297ff17b100fb0d61c4a70e5fef4d77c2c/sky/cli.py#L241-L247